src/page-analysis/index.js

```diff
@@ -2,24 +2,22 @@
 
 // The properties of a page that should be indexed for full-text search.
 export const searchableTextFields = [
+    'fullText',
     'title',
-    'extractedMetadata.title',
-    'extractedMetadata.Title',
-    'extractedMetadata.Author',
-    'extractedMetadata.Subject',
-    'extractedMetadata.Keywords',
-    'extractedText.excerpt',
-    'extractedText.textContent',
-    'extractedText.bodyInnerText',
+    'author',
+    'description',
+    'keywords',
 ]
 
-// Revise and augment a page doc using available fields from the extracted data.
-// (done at retrieval rather than at storage time for the sake of flexibility)
+// Revise and augment a page doc, used for two possible reasons:
+//   1: Rename old fields to the current scheme, without needing data migration.
+//   2: Choose the best fields from the available data (doing this at retrieval
+//      rather than at storage time gives flexibility).
 export const revisePageFields = doc => ({
     ...doc,
-    // Choose something presentable as a title.
-    title: (doc.extractedMetadata && doc.extractedMetadata.title)
-        || (doc.extractedMetadata && doc.extractedMetadata.Title)
-        || doc.title
-        || doc.url,
+    title: (
+        doc.title
+        || (doc.extractedMetadata && doc.extractedMetadata.title) // old format
+        || (doc.extractedMetadata && doc.extractedMetadata.Title) // old format
+    ),
```

Comment:
I might remove these lines again either now or later. Being backwards compatible with a data model that was never really published sounds like unnecessary complexity again.